src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs

```diff
@@ -535,6 +532,58 @@ public override void ProjectStartedHandler(object sender, ProjectStartedEventArg
                     WriteItems(e, e.Items);
                 }
             }
+
+            //node and project context ids for the propertyOutputMap key
+            int node = -1;
+            int projectContextId = -1;
+            if (e.BuildEventContext != null)
+            {
+                node = e.BuildEventContext.NodeId;
+                projectContextId = e.BuildEventContext.ProjectContextId;
+            }
+            //creating the value to be added to the propertyOutputMap
+            StringBuilder LogOutputProperties = new StringBuilder();
+            if (e.BuildEventContext != null && e.Items != null)
+            {
+                foreach (DictionaryEntry item in e.Items)
+                {
+                    ITaskItem itemVal = (ITaskItem)item.Value;
+                    //finding if the LogOutputProperties item has been used
+                    if (string.Equals((string)item.Key, "LogOutputProperties", StringComparison.OrdinalIgnoreCase))
+                    {
+                        //looking for the property value associated with the property key
+                        //Note: the property key is the item value
+                        string value = null;
+                        bool foundProperty = e.GlobalProperties.TryGetValue(itemVal.ItemSpec, out value);
+     
+                        //looking for the property in local properties if it wasn't found in global properties
+                        if (!foundProperty)
+                        {
+                            foreach (DictionaryEntry prop in e.Properties)
+                            {
+
+                                if ((string)prop.Key == itemVal.ItemSpec)
+                                {
+                                    value = (string)prop.Value;
+                                    foundProperty = true;
+                                    break;
+                                }
+                            }
+                        }
+
+                        //adding the property key and value pair to the propertyOutputs
+                        if (e.BuildEventContext != null && foundProperty)
+                        {
+                            LogOutputProperties.Append(itemVal.ItemSpec).Append(":").Append(value).Append(" ");
+                        }
+                    }
+                }
+            }
+            //adding the finished dictionary to propertyOutputMap
+            //this creates a mapping of a specific project/node to a dictionary of property values
+            if (e.BuildEventContext != null)
+                if (!propertyOutputMap.ContainsKey((node, projectContextId)))
```

Comment:
Under what circumstances does `propertyOutputMap` already contain something for this?

Reply:
That's a good point. It's an unnecessary check I can remove